### ISSUE TITLE
Broken label association: for="userid" does not match id="login" (html accessibility)

### ISSUE THREAD

**Issue by loganripplinger** [dexidp/dex#4540]
### Preflight Checklist

- [x] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.
- [x] I am not looking for support or already pursued the available [support channels](https://github.com/dexidp/dex/issues/new/choose) without success.

### Version

master

### Storage Type

etcd

### Installation Type

Official container image

### Expected Behavior

When using the default login template:
- Clicking the username label focuses the username input.

The label should have correct accessibility association with the input for assistive technologies /  Playwright automated tests.

### Actual Behavior

- Clicking the username label does not focus the username input.
    

### Steps To Reproduce

1. Load the login page.
2. Click the “Username” label.
3. Observe that focus does not move to the username input.

Do the same with "Password" label and observe that the input grabs focus.

### Additional Information

Code: https://github.com/dexidp/dex/blob/489e37d07ad987adc1152897de512202941304fb/web/templates/password.html#L8-L10

### Summary

The username `<label>`’s `for` attribute references `userid`, but the corresponding `<input>` has `id="login"`. As a result, clicking the “Username” label does not focus the username field, assistive technology may not correctly associate the label with the input, and automated testing tools (ex: Playwright) cannot reliably use `getByLabelText()` / `byLabelText`-style selectors.

**Root cause:**

```html
<label for="userid">{{ .UsernamePrompt }}</label>
<input tabindex="1" required id="login" name="login" type="text" ... />
```

The `for` attribute references `userid`, but the input’s `id` is `login`.

#### Proposed fix (minimal / non-breaking):

```html
<label for="login">{{ .UsernamePrompt }}</label>
```

This preserves the existing input `id` and `name`, minimizing risk to CSS, JS, password managers, or external integrations while restoring correct accessibility and automation behavior.

### Configuration

```yaml

```

### Logs

```shell

```
